src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -775,6 +841,30 @@ bool primitive_inst::update_impl() {
     return true;
 }
 
+void primitive_inst::init_variable() {
```

Comment:
Maybe this function should be renamed to `update_paddings()` or `init_paddings()` as implementation actually doesn't init variable? In that case 
```cpp
    if (_node->is_type<gather>() && _impl_params->output_layouts[0].data_padding.get_dynamic_pad_dims() != tensor(0)) {
        if (can_be_optimized())
            _impl_params->output_layouts[0] = _impl_params->input_layouts[0];
        else
            reset_pad(*_impl_params, _node);
        return;
    }
```
part can also be moved into this function

Reply:
Fixed as suggestion